src/main/scala/treadle/stage/phases/GetFirrtlAst.scala

```diff
@@ -29,8 +31,15 @@ object GetFirrtlAst extends Phase {
     def handleFirrtlSource(): Option[AnnotationSeq] = {
       annotationSeq.collectFirst { case FirrtlSourceAnnotation(firrtlText) => firrtlText } match {
         case Some(text) =>
-          val circuit = Parser.parse(text)
+
+          //Run the additional coverage firrtl passes if needed
+          val circuit = if (annotationSeq.contains(EnableCoverageAnnotation)) {
```

Comment:
This should probably be done through a `RunFirrtlTransformAnnotation` instead of checking for your annotation manually. @chick probably has a more precise answer on how to integrate this with the existing treadle infrastructure.

Reply:
I think Kevin is right here and I think it's pretty simple to change.
If you do something like the following
Add this to AddCoverageExpressions
```
  override def optionalPrerequisiteOf: Seq[TransformDependency] = Seq(Dependency[BlackBoxSourceHelper])
```
Add 
```
    Dependency(AddCoverageExpressions),
```
to line 18 of PrepareAst, then revert the changes to `GetFirrtlAst` as the other changes will have scheduled your pass.